### ISSUE TITLE
Type Error in `@xixixao/uploadstuff` - `useEvent.ts:38:21` 

### ISSUE THREAD

**Issue by RicardoGEsteves** [get-convex/uploadstuff#16]
**Description:**  
A type error occurs in the `@xixixao/uploadstuff` package, specifically in the `useEvent.ts` file, when attempting to build or run the application. The error message indicates that a `useRef` call is missing a required argument.  

**Error Details:**  

```
./node_modules/.pnpm/@xixixao+uploadstuff@0.0.5_react-dropzone@14.3.5_react@19.0.0__react@19.0.0_typescript@5.7.2/node_modules/@xixixao/uploadstuff/lib/useEvent.ts:38:21  
Type error: Expected 1 arguments, but got 0.

  36 |   // Create a stable callback that always calls the latest callback:
  37 |   // using useRef instead of useCallback avoids creating and empty array on every render
> 38 |   const stableRef = useRef<TCallback>();
     |                     ^
  39 |   if (!stableRef.current) {
  40 |     stableRef.current = function (this: unknown) {
  41 |       // eslint-disable-next-line @typescript-eslint/no-unsafe-return, prefer-rest-params, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-explicit-any
```

<img width="1005" alt="Screenshot 2024-12-13 at 21 32 03" src="https://github.com/user-attachments/assets/bf743312-be12-4b9a-86a3-72dd2ba69773" />

**Steps to Reproduce:**  
1. Install the `@xixixao/uploadstuff` package, version `0.0.5`, with dependencies including:  
   - `react-dropzone@14.3.5`  
   - `react@19.0.0`  
   - `typescript@5.7.2`  
2. Attempt to build or run the project.  
3. Observe the type error in the `useEvent.ts` file at line 38.  

**Expected Behavior:**  
The code should compile and run without type errors.  

**Environment:**  
- Node.js version: 22 
- npm version: 10 
- `pnpm` version: 9.15 
- Operating System: macOS Sequoia 15.1.1

**Additional Information:**  
The issue seems related to the usage of `useRef` without an initial argument, which is required for proper typing in TypeScript. Providing a default value or ensuring compatibility with the expected type might resolve the issue.  

**Simple Fix:**
```
const stableRef = useRef<TCallback | null>(null);
```

Would be nice to fix this without needing to patch the module.

**Comment by xixixao** [comment]
This is because the types changed in React, and indeed should be easy to fix. Feel free to open a PR!

**Comment by RicardoGEsteves** [comment]
> This is because the types changed in React, and indeed should be easy to fix. Feel free to open a PR!

Yeah I know now useRef expects/requires always an argument, similar to createContext.
Will do, will raise a PR with the proposed fix.

**Comment by RicardoGEsteves** [comment]
Hi @xixixao, raised a `PR` already [fix(useRef): initialize stableRef with null to comply with React 19 requirements](https://github.com/get-convex/uploadstuff/pull/18) , when you have time can you have a look? should be really straightforward.

**Comment by ChenBingWei1201** [comment]
Hi @RicardoGEsteves, I have encountered the same issue but I wonder why you closed your `pr` (#17 and #18 )

**Comment by RicardoGEsteves** [comment]
Hi @ChenBingWei1201, well was open for quite some time I ever refer it on the issues but didn't got any reply so I decided to implement/create my own uploader.

**Comment by reanzi** [comment]
That's so sad , on my local dev, passing null solved the issue, only to appear again in production; 
@RicardoGEsteves do you have a repo to your upoader you created? I'll appreciate a help here
